infrastructure/removePublicGCS/index.js

```diff
@@ -0,0 +1,23 @@
+/**
+ * Background Cloud Function to be triggered by Cloud Storage.
+ *
+ * @param {object} event The Cloud Functions event.
+ * @param {function} callback The callback function.
+ */
+
+const gcs = require('@google-cloud/storage')();
+
+exports.removePublicAccess = function (event, callback) {
+  const object = event.data;
+  const file = gcs.bucket(object.bucket).file(object.name);
+  console.log(`Ensuring no public ACLs on ${file.name}`);
+  file.acl.owners.deleteAllAuthenticatedUsers();
```

Comment:
This pattern needs a few elements of cleanup.

* The deleteFoo() methods are blind - each of those will issue an API call whether or not the ACL includes the unwanted users. In the `event` object you should be receiving the resource representation of the Object, including the ACLS (https://cloud.google.com/storage/docs/json_api/v1/objects#resource-representations). You can greatly reduce the relative cost of this function by checking the ACLs for the presence of 'allUsers' or 'allAuthenticatedUsers' before invoking.
* Each of these function calls is returning a Promise since there's no callback provided, but you're dropping the Promise. You should issue your API calls in parallel like this, but wait for them all to return - see examples here: https://coderwall.com/p/ijy61g/promise-chains-with-node-js
* You need to have some error logging in the event your call to modify the ACL doesn't work - see above about checking for the results of the promises from your API calls.